src/Microsoft.AspNet.OData.Shared/Builder/LinkGenerationHelpers.cs

```diff
@@ -95,6 +95,83 @@ public static Uri GenerateSelfLink(this ResourceContext resourceContext, bool in
             return new Uri(link);
         }
 
+        /// <summary>
+        /// Generates a contained navigation link following the OData URL conventions for the entity represented by <paramref name="resourceContext"/> and the given
+        /// navigation property.
+        /// </summary>
+        /// <param name="resourceContext">The <see cref="ResourceContext"/> representing the entity for which the navigation link needs to be generated.</param>
+        /// <param name="navigationProperty">The EDM navigation property.</param>
+        /// <returns>The navigation link following the OData URL conventions.</returns>
+        public static Uri GenerateContainedNavigationPropertyLink(this ResourceContext resourceContext,
```

Comment:
>GenerateContainedNavigationPropertyLink [](start = 26, length = 39)

Can we use the above "GenerateNavigationPropertyLink" method with some changes?

Reply:
Added `includeCast`
#Resolved